server/master/src/main/java/org/apache/accumulo/master/tableOps/PopulateMetadata.java

```diff
@@ -42,8 +43,11 @@ public long isReady(long tid, Master environment) throws Exception {
     MetadataTableUtil.addTablet(extent, tableInfo.dir, environment, tableInfo.timeType,
         environment.getMasterLock());
 
-    return new FinishCreateTable(tableInfo);
-
+    if (tableInfo.props.containsKey(Property.TABLE_OFFLINE_OPTS + "create.initial.splits")) {
```

Comment:
I cloned this branch to play around with moving adding splits into this class.   I got as far as writing the following function.  I think this function would work for 0 or more splits.  One thing I realized while attempting this exercise is the current code is using the same dir for each tablet which is not ideal (it may work).  So `dirFunction` below represents this TODO.

```java
  private void writeSplitsToMetadataTable(Table.ID tableId, SortedMap<Text> splits, Function<Text,Value> dirFunction, char timeType,
      ZooLock lock, BatchWriter bw) {
    Text prevSplit = null;
    for(Text split : Iterables.concat(splits, Collections.singleton(null))) {
      Mutation mut = new KeyExtent(tableId, split, prevSplit).getPrevRowUpdateMutation();
      ServerColumnFamily.DIRECTORY_COLUMN.put(mut, dirFunction.apply(split));
      ServerColumnFamily.TIME_COLUMN.put(mut, new Value(timeType+"0"));
      MetadataTableUtil.putLockID(lock, mut);
      prevSplit = split;

      bw.addMutation(mut);
    }
  }
```

One possible way to handle this is to do the following.

 * In the ChooseDir FATE step, read in the splits and choose a dir for each split.  Write out a new file with the dirs.
 * In the CreateDir FATE step, read the file with dirs and create them.
 * In the PopulateMetadata FATE step, read the file with dirs and use those for creating rows in metadata table.